app/views/edit/build-config.html

```diff
@@ -20,57 +20,90 @@
                   <div class="row">
-                    <div class="col-lg-6">
+                    <div ng-class="{
+                          'col-lg-6': advancedOptions,
+                          'col-lg-12': !advancedOptions}">
                       <div ng-if="buildConfig.spec.source.type !== 'None'" class="section">
                         <h3>Source Configuration</h3>
                         <dl class="dl-horizontal left">
                           <div ng-if="sources.git">
-                            <div class="form-group">
-                              <label for="sourceUrl">Source Repository URL</label>
-                              <div>
-                                <!-- Unfortunately, we can't set type="url" because some valid values don't pass browser validation. -->
-                                <input class="form-control"
-                                  id="sourceUrl"
-                                  name="sourceUrl"
-                                  ng-model="updatedBuildConfig.spec.source.git.uri"
-                                  type="text"
-                                  autocorrect="off"
-                                  autocapitalize="off"
-                                  spellcheck="false"
-                                  required>
+                            <div class="row">
+                              <div class="col-lg-9">
+
+                                <div class="form-group">
+                                  <label for="sourceUrl" class="required">Git Repository URL</label>
+                                  <div>
+                                    <!-- Unfortunately, we can't set type="url" because some valid values don't pass browser validation. -->
+                                    <input class="form-control"
+                                      id="sourceUrl"
+                                      name="sourceUrl"
+                                      ng-model="updatedBuildConfig.spec.source.git.uri"
+                                      type="text"
+                                      autocorrect="off"
+                                      autocapitalize="off"
+                                      spellcheck="false"
+                                      required>
+                                  </div>
+                                  <div>
+                                    <span class="text-warning" ng-if="form.sourceUrl.$dirty && !sourceURLPattern.test(updatedBuildConfig.spec.source.git.uri)">Git repository should be a URL.</span>
+                                  </div>
+                                </div>
+
                               </div>
-                              <div>
-                                <span class="text-warning" ng-if="form.sourceUrl.$dirty && !sourceURLPattern.test(updatedBuildConfig.spec.source.git.uri)">Git repository should be a URL.</span>
+                              <div class="col-lg-3">
```

Comment:
I'm not convinced this is better than having git reference as a separate field below.

I think we should hide "Git Reference" until show advanced options is clicked.